developer-docs-site/docs/reference/move.md

```diff
@@ -0,0 +1,155 @@
+* A [timestamp service](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/Timestamp.move) that provides a monotonically increasing clock that maps to the actual current unixtime
+
+With much more coming soon...
+
+## Key Concepts in Move
+
+* Data should be stored within the account that owns it not the account that published the module.
+* Data flow should have minimal constraints with an emphasis on ecosystem usability
+* Prefer static type-safety over run-time safety via generics
+* A `signer` should be required to restrict access to adding or removing assets to an account unless it is explicitly clear
+
+### Data ownership
+
+Data should be stored within the account that owns it not the account that published the module.
+
+In Solidity, data is stored within the namespace of the account that created the contract. Typically this is represented by a map of an address to a value or of an instance id to the address of the owner.
+
+In Solana, data is stored within a distinct account associated with a contract.
+
+In Move, data can be stored within the module owner's account, but that creates the issue of ownership ambiguity and implies that two issues:
+1. It makes ownership ambiguous as the asset has no resource associated with the owner
+2. The module creator takes responsibility for the lifetime of that resources, e.g., rent, reclamation, etc
+
+On the first point, by placing assets within trusted resources within an account, the owner can ensure that even a maliciously programmed module will be unable to modify those assets. In Move, we can program a standard orderbook structure and interface that would let applications built on top be unable to gain backdoor access to an account or its orderbook entries.
+
+Contrast the following two coin storage strategies:
+
+The following places the coins into a single account with ownership indicated by an index:
+```rust
+struct CoinStore has key {
+    coins: table<address, Coin>,
+}
+```
+
+Instead prefer the approach that stores the coins in an account:
+```rust
+struct CoinStore has key {
+    coins: Coin,
+}
+```
+
+This makes ownership explicit.
+
+### Data flow
+
+Data flow should have minimal constraints with an emphasis on ecosystem usability
+
+Assets can be programmed to be entirely constrained within a module by making it such that no interface ever presents the struct in a value form and instead only provides functions for manipulating the data defined within the module. This constrains the data's availability to only within a module and makes it unexportable, which in turn prevents interoperability with other modules. Specifically, one could imagine a purchase contract that takes as input some `Coin<T>` and returns a `Ticket`. If `Coin<T>` is only defined within the module and is not exportable outside, then the applications for that `Coin<T>` are limited to whatever the module has defined.
```

Comment:
One module cannot directly access another module's resource. It can only manipulate other modules' resource through the modules' public functions.